File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.java


import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int result = 0;

        for(int n=0;n<N;n++) {
            boolean isGroup = true;

            String word = br.readLine();
            char[] cArray = word.toCharArray();

            for(char c: cArray) {
                int startIndex = word.indexOf(c);
                int endIndex = word.lastIndexOf(c);

                if(startIndex!=endIndex) {
                    for(int middleIndex=startIndex;middleIndex<endIndex;middleIndex++){
                        if(word.charAt(middleIndex)!=c) {
                            isGroup = false;
                            break;
                        }
                    }
                }
            }
            if(isGroup) {
                result++;
            }
        }

        System.out.println(result);
    }
}

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.py
import sys
input = sys.stdin.readline

N = int(input().strip())
result = N
for i in range(N):
    word = input().strip()
    my_set = set()
    before = word[0]
    for c in word:
        if(c in my_set):
            if(c==before):
                continue
            else:
                result -= 1
                break
        else:
            my_set.add(c)
            before = c
print(result)